tuning/sft_trainer.py

```diff
@@ -317,7 +318,9 @@ def train(
         formatted_train_dataset,
         formatted_validation_dataset,
         dataset_text_field,
-    ) = format_dataset(data_args, tokenizer, max_seq_length)
+        #    ) = format_dataset(data_args, tokenizer, max_seq_length)
+    ) = process_dataargs(data_args, tokenizer, max_seq_length)
```

Comment:
this seems to be the only entrypoint into sft_trainer. But I cant understand how something like this can switch out the dataloader.

Reply:
@fabianlim fair point but this is because the patch is not complete yet..its `WIP` and I will add additional commandline args to take a data config like the one in ADR from the user. 
The patch was made public following our internal discussion to have @willmj  and @Abhishek-TAMU visibility into the implementation and helping with it if needed.